The job `x86_64-gnu-llvm-7` of your PR [failed](https://dev.azure.com/rust-lang/rust/_build/results?buildId=23223&view=logs&jobId=0230d20d-e0df-5b62-a6ec-fad0f9c3c3d5) ([pretty log](https://rust-lang.github.io/rust-log-analyzer/log-viewer/#https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/23223/logs/71), [raw log](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/23223/logs/71)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
2020-03-14T12:59:16.0627738Z ========================== Starting Command Output ===========================
2020-03-14T12:59:16.0631928Z [command]/bin/bash --noprofile --norc /home/vsts/work/_temp/557bc36b-c2fe-4be8-a01f-b86018ff64f4.sh
2020-03-14T12:59:16.0632195Z 
2020-03-14T12:59:16.0636323Z ##[section]Finishing: Disable git automatic line ending conversion
2020-03-14T12:59:16.0656311Z ##[section]Starting: Checkout rust-lang/rust@refs/pull/70000/merge to s
2020-03-14T12:59:16.0659450Z Task         : Get sources
2020-03-14T12:59:16.0659756Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
2020-03-14T12:59:16.0660035Z Version      : 1.0.0
2020-03-14T12:59:16.0660282Z Author       : Microsoft
---
2020-03-14T12:59:17.0531370Z ##[command]git remote add origin https://github.com/rust-lang/rust
2020-03-14T12:59:17.0536602Z ##[command]git config gc.auto 0
2020-03-14T12:59:17.0540294Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2020-03-14T12:59:17.0543724Z ##[command]git config --get-all http.proxy
2020-03-14T12:59:17.0549780Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/70000/merge:refs/remotes/pull/70000/merge
---
2020-03-14T14:00:58.2718325Z .................................................................................................... 1700/9772
2020-03-14T14:01:02.4153343Z .................................................................................................... 1800/9772
2020-03-14T14:01:13.7631395Z ...................................................................i................................ 1900/9772
2020-03-14T14:01:20.4981381Z .................................................................................................... 2000/9772
2020-03-14T14:01:34.7563408Z .........................................................iiiii...................................... 2100/9772
2020-03-14T14:01:45.3562844Z .................................................................................................... 2300/9772
2020-03-14T14:01:47.5543870Z .................................................................................................... 2400/9772
2020-03-14T14:01:50.5498747Z .................................................................................................... 2500/9772
2020-03-14T14:02:12.4145057Z .................................................................................................... 2600/9772
---
2020-03-14T14:04:44.9851067Z .............................i...............i...................................................... 5000/9772
2020-03-14T14:04:54.1210071Z .................................................................................................... 5100/9772
2020-03-14T14:04:59.9384559Z ........................................................................i........................... 5200/9772
2020-03-14T14:05:05.2730792Z .................................................................................................... 5300/9772
2020-03-14T14:05:14.6574435Z .....................................................ii.ii........i...i............................. 5400/9772
2020-03-14T14:05:22.4627698Z .................................................................................................... 5600/9772
2020-03-14T14:05:31.7060229Z .................................................................................................... 5700/9772
2020-03-14T14:05:37.5368605Z .............................................i...................................................... 5800/9772
2020-03-14T14:05:43.7439842Z .................................................................................................... 5900/9772
2020-03-14T14:05:43.7439842Z .................................................................................................... 5900/9772
2020-03-14T14:05:53.3737014Z .................................................................................................... 6000/9772
2020-03-14T14:05:59.1355240Z .......................................ii...i..ii...........i....................................... 6100/9772
2020-03-14T14:06:18.4142099Z .................................................................................................... 6300/9772
2020-03-14T14:06:24.9666881Z .................................................................................................... 6400/9772
2020-03-14T14:06:24.9666881Z .................................................................................................... 6400/9772
2020-03-14T14:06:33.9371739Z .....................................................................i..ii.......................... 6500/9772
2020-03-14T14:06:57.1628039Z .................................................................................................... 6700/9772
2020-03-14T14:07:05.3489942Z ...................................................................i................................ 6800/9772
2020-03-14T14:07:07.2691169Z .................................................................................................... 6900/9772
2020-03-14T14:07:09.3091223Z .................................................................................................... 7000/9772
---
2020-03-14T14:08:49.1852743Z .................................................................................................... 7800/9772
2020-03-14T14:08:54.9103152Z .................................................................................................... 7900/9772
2020-03-14T14:09:00.7448075Z ....................................................i............................................... 8000/9772
2020-03-14T14:09:10.9795632Z .................................................................................................... 8100/9772
2020-03-14T14:09:16.1302950Z .iiiiiiiiii.i....................................................................................... 8200/9772
2020-03-14T14:09:29.2798856Z .................................................................................................... 8400/9772
2020-03-14T14:09:40.0594542Z .................................................................................................... 8500/9772
2020-03-14T14:09:52.3217056Z .................................................................................................... 8600/9772
2020-03-14T14:09:57.9068525Z .................................................................................................... 8700/9772
---
2020-03-14T14:12:11.1129622Z  finished in 7.110
2020-03-14T14:12:11.1309165Z Check compiletest suite=codegen mode=codegen (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2020-03-14T14:12:11.2858996Z 
2020-03-14T14:12:11.2859336Z running 181 tests
2020-03-14T14:12:13.9695536Z iiii......i............ii..iiii....i....i...........i............i..i..................i....i....... 100/181
2020-03-14T14:12:16.4939042Z .....i.i.i...iii..iiiiiiiiiiiiiiii.......................iii.............ii......
2020-03-14T14:12:16.4941694Z 
2020-03-14T14:12:16.4946626Z  finished in 5.365
2020-03-14T14:12:16.5132080Z Check compiletest suite=codegen-units mode=codegen-units (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2020-03-14T14:12:16.6667714Z 
---
2020-03-14T14:12:18.4678217Z  finished in 1.953
2020-03-14T14:12:18.4844339Z Check compiletest suite=assembly mode=assembly (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2020-03-14T14:12:18.6380411Z 
2020-03-14T14:12:18.6381069Z running 9 tests
2020-03-14T14:12:18.6382295Z iiiiiiiii
2020-03-14T14:12:18.6383719Z 
2020-03-14T14:12:18.6388866Z  finished in 0.154
2020-03-14T14:12:18.6566803Z Check compiletest suite=incremental mode=incremental (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2020-03-14T14:12:18.8139916Z 
---
2020-03-14T14:12:37.7374399Z  finished in 19.080
2020-03-14T14:12:37.7559705Z Check compiletest suite=debuginfo mode=debuginfo (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2020-03-14T14:12:37.9149843Z 
2020-03-14T14:12:37.9150178Z running 115 tests
2020-03-14T14:12:50.9087102Z iiiii..i.....i..i...i..i.i.i..i..i..ii....i.i....ii..........iiii.........i.....i..i.......ii.i.ii.. 100/115
2020-03-14T14:12:52.6655712Z ...iiii.....ii.
2020-03-14T14:12:52.6656955Z 
2020-03-14T14:12:52.6658549Z  finished in 14.910
2020-03-14T14:12:52.6664612Z Uplifting stage1 rustc (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2020-03-14T14:12:52.6667514Z Copying stage2 rustc from stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu / x86_64-unknown-linux-gnu)
---
2020-03-14T14:25:47.2026447Z 
2020-03-14T14:25:47.2031008Z    Doc-tests core
2020-03-14T14:25:51.9389286Z 
2020-03-14T14:25:51.9390425Z running 2480 tests
2020-03-14T14:26:00.7604648Z ......iiiii......................................................................................... 100/2480
2020-03-14T14:26:09.4773421Z ....................................................................................ii.............. 200/2480
2020-03-14T14:26:29.4955561Z ...................i................................................................................ 400/2480
2020-03-14T14:26:29.4955561Z ...................i................................................................................ 400/2480
2020-03-14T14:26:38.8051251Z ........................................................................i..i..................iiii.. 500/2480
2020-03-14T14:26:54.6282023Z .................................................................................................... 700/2480
2020-03-14T14:27:02.8351568Z .................................................................................................... 800/2480
2020-03-14T14:27:10.8706560Z .................................................................................................... 900/2480
2020-03-14T14:27:19.0365735Z .................................................................................................... 1000/2480
---
2020-03-14T14:30:47.9304736Z 
2020-03-14T14:30:47.9306512Z running 1010 tests
2020-03-14T14:31:05.7015720Z i................................................................................................... 100/1010
2020-03-14T14:31:15.7579199Z .................................................................................................... 200/1010
2020-03-14T14:31:22.8128370Z ..................ii.i.....i......i...i......i...................................................... 300/1010
2020-03-14T14:31:27.7530205Z .................................................................................................... 400/1010
2020-03-14T14:31:34.3328204Z ............................................i..i......................................ii............ 500/1010
2020-03-14T14:31:46.4320873Z .................................................................................................... 700/1010
2020-03-14T14:31:46.4320873Z .................................................................................................... 700/1010
2020-03-14T14:31:53.0160768Z ....................................iiii............................................................ 800/1010
2020-03-14T14:32:06.8519331Z .................................................................................................... 900/1010
2020-03-14T14:32:13.4756885Z ..........................................................iiii...................................... 1000/1010
2020-03-14T14:32:13.9156333Z test result: ok. 990 passed; 0 failed; 20 ignored; 0 measured; 0 filtered out
2020-03-14T14:32:13.9156588Z 
2020-03-14T14:32:13.9274256Z  finished in 163.432
2020-03-14T14:32:13.9288705Z Testing term stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
---
2020-03-14T14:50:17.7371378Z  finished in 38.872
2020-03-14T14:50:17.7629785Z Check compiletest suite=run-make-fulldeps mode=run-make (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2020-03-14T14:50:17.9720665Z 
2020-03-14T14:50:17.9722284Z running 210 tests
2020-03-14T14:50:48.9234682Z ......................i...ii.......................................................................i 100/210
2020-03-14T14:51:25.3761525Z ........................................iiiiii......i..............iii.............................. 200/210
2020-03-14T14:51:29.8685048Z test result: ok. 195 passed; 0 failed; 15 ignored; 0 measured; 0 filtered out
2020-03-14T14:51:29.8685323Z 
2020-03-14T14:51:29.8690480Z  finished in 72.106
2020-03-14T14:51:29.8699060Z doc tests for: /checkout/src/doc/rustdoc/src/advanced-features.md
---
2020-03-14T14:51:57.6287903Z 
2020-03-14T14:51:57.6288348Z ---- [ui] rustdoc-ui/issue-61732.rs stdout ----
2020-03-14T14:51:57.6288581Z diff of stderr:
2020-03-14T14:51:57.6288736Z 
2020-03-14T14:51:57.6289233Z - error[E0577]: expected module, found unresolved item `crate::r#mod`
2020-03-14T14:51:57.6289741Z -   --> $DIR/issue-61732.rs:3:8
2020-03-14T14:51:57.6290063Z + error[E0433]: failed to resolve: maybe a missing crate `r#mod`?
2020-03-14T14:51:57.6290772Z 3    |
2020-03-14T14:51:57.6290772Z 3    |
2020-03-14T14:51:57.6290984Z 4 LL | pub(in crate::r#mod) fn main() {}
2020-03-14T14:51:57.6291434Z -    |        ^^^^^^^^^^^^ not a module
2020-03-14T14:51:57.6291724Z +    |               ^^^^^ maybe a missing crate `r#mod`?
2020-03-14T14:51:57.6292161Z 7 error: Compilation failed, aborting rustdoc
2020-03-14T14:51:57.6292361Z 8 
2020-03-14T14:51:57.6292464Z 
2020-03-14T14:51:57.6292656Z 9 error: aborting due to 2 previous errors
---
2020-03-14T14:51:57.6294236Z 
2020-03-14T14:51:57.6294333Z 
2020-03-14T14:51:57.6294540Z The actual stderr differed from the expected stderr.
2020-03-14T14:51:57.6295215Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/issue-61732/issue-61732.stderr
2020-03-14T14:51:57.6303064Z To update references, rerun the tests and pass the `--bless` flag
2020-03-14T14:51:57.6303716Z To only update this specific test, also pass `--test-args issue-61732.rs`
2020-03-14T14:51:57.6304181Z error: 1 errors occurred comparing output.
2020-03-14T14:51:57.6304423Z status: exit code: 1
2020-03-14T14:51:57.6304423Z status: exit code: 1
2020-03-14T14:51:57.6306353Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc" "/checkout/src/test/rustdoc-ui/issue-61732.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/issue-61732" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/issue-61732/auxiliary"
2020-03-14T14:51:57.6307944Z ------------------------------------------
2020-03-14T14:51:57.6308121Z 
2020-03-14T14:51:57.6308480Z ------------------------------------------
2020-03-14T14:51:57.6308701Z stderr:
2020-03-14T14:51:57.6308701Z stderr:
2020-03-14T14:51:57.6309075Z ------------------------------------------
2020-03-14T14:51:57.6309398Z error[E0433]: failed to resolve: maybe a missing crate `r#mod`?
2020-03-14T14:51:57.6310188Z    |
2020-03-14T14:51:57.6310188Z    |
2020-03-14T14:51:57.6310392Z LL | pub(in crate::r#mod) fn main() {}
2020-03-14T14:51:57.6310682Z    |               ^^^^^ maybe a missing crate `r#mod`?
2020-03-14T14:51:57.6311094Z error: Compilation failed, aborting rustdoc
2020-03-14T14:51:57.6311271Z 
2020-03-14T14:51:57.6311458Z error: aborting due to 2 previous errors
2020-03-14T14:51:57.6311645Z 
---
2020-03-14T14:51:57.6314981Z thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:348:22
2020-03-14T14:51:57.6315397Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
2020-03-14T14:51:57.6324184Z 
2020-03-14T14:51:57.6324342Z 
2020-03-14T14:51:57.6328452Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--rustdoc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc" "--src-base" "/checkout/src/test/rustdoc-ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-7/bin/FileCheck" "--host-rustcflags" "-Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "7.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
2020-03-14T14:51:57.6331304Z 
2020-03-14T14:51:57.6331405Z 
2020-03-14T14:51:57.6331658Z failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
2020-03-14T14:51:57.6331991Z Build completed unsuccessfully in 1:45:54
2020-03-14T14:51:57.6331991Z Build completed unsuccessfully in 1:45:54
2020-03-14T14:51:57.6365047Z == clock drift check ==
2020-03-14T14:51:57.6380752Z   local time: Sat Mar 14 14:51:57 UTC 2020
2020-03-14T14:51:57.9227937Z   network time: Sat, 14 Mar 2020 14:51:57 GMT
2020-03-14T14:51:57.9228347Z == end clock drift check ==
2020-03-14T14:51:58.4732133Z 
2020-03-14T14:51:58.4811904Z ##[error]Bash exited with code '1'.
2020-03-14T14:51:58.4826672Z ##[section]Finishing: Run build
2020-03-14T14:51:58.4880117Z ##[section]Starting: Checkout rust-lang/rust@refs/pull/70000/merge to s
2020-03-14T14:51:58.4885770Z Task         : Get sources
2020-03-14T14:51:58.4886113Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
2020-03-14T14:51:58.4886447Z Version      : 1.0.0
2020-03-14T14:51:58.4886669Z Author       : Microsoft
2020-03-14T14:51:58.4886669Z Author       : Microsoft
2020-03-14T14:51:58.4887023Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=798199)
2020-03-14T14:51:58.4887446Z ==============================================================================
2020-03-14T14:51:58.8307980Z Cleaning any cached credential from repository: rust-lang/rust (GitHub)
2020-03-14T14:51:58.8356259Z ##[section]Finishing: Checkout rust-lang/rust@refs/pull/70000/merge to s
2020-03-14T14:51:58.8446288Z Cleaning up task key
2020-03-14T14:51:58.8447711Z Start cleaning up orphan processes.
2020-03-14T14:51:58.8639995Z Terminate orphan process: pid (4292) (python)
2020-03-14T14:51:58.8888562Z ##[section]Finishing: Finalize Job
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@rust-lang/infra`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

--- a/src/librustc_resolve/lib.rs
+++ b/src/librustc_resolve/lib.rs
@@ -2183,11 +2183,8 @@ impl<'a> Resolver<'a> {
                                     Applicability::MaybeIncorrect,
                                 )),
                             )
-                        } else if !ident.is_reserved() {
-                            (format!("maybe a missing crate `{}`?", ident), None)
                         } else {
-                            // the parser will already have complained about the keyword being used
-                            return PathResult::NonModule(PartialRes::new(Res::Err));
+                            (format!("maybe a missing crate `{}`?", ident), None)
                         }
                     } else if i == 0 {
                         (format!("use of undeclared type or module `{}`", ident), None)

--- a/src/test/rustdoc-ui/issue-61732.rs
+++ b/src/test/rustdoc-ui/issue-61732.rs
@@ -1,4 +1,4 @@
 // This previously triggered an ICE.
 
 pub(in crate::r#mod) fn main() {}
-//~^ ERROR expected module, found unresolved item
+//~^ ERROR failed to resolve: maybe a missing crate `r#mod`

--- a/src/test/rustdoc-ui/issue-61732.stderr
+++ b/src/test/rustdoc-ui/issue-61732.stderr
@@ -1,11 +1,11 @@
-error[E0577]: expected module, found unresolved item `crate::r#mod`
-  --> $DIR/issue-61732.rs:3:8
+error[E0433]: failed to resolve: maybe a missing crate `r#mod`?
+  --> $DIR/issue-61732.rs:3:15
    |
 LL | pub(in crate::r#mod) fn main() {}
-   |        ^^^^^^^^^^^^ not a module
+   |               ^^^^^ maybe a missing crate `r#mod`?
 
 error: Compilation failed, aborting rustdoc
 
 error: aborting due to 2 previous errors
 
-For more information about this error, try `rustc --explain E0577`.
+For more information about this error, try `rustc --explain E0433`.

--- a/src/test/ui/keyword/extern/keyword-extern-as-identifier-use.rs
+++ b/src/test/ui/keyword/extern/keyword-extern-as-identifier-use.rs
@@ -1,3 +1,4 @@
 use extern::foo; //~ ERROR expected identifier, found keyword `extern`
+                 //~| ERROR unresolved import `r#extern`
 
 fn main() {}

--- a/src/test/ui/keyword/extern/keyword-extern-as-identifier-use.stderr
+++ b/src/test/ui/keyword/extern/keyword-extern-as-identifier-use.stderr
@@ -9,5 +9,12 @@ help: you can escape reserved keywords to use them as identifiers
 LL | use r#extern::foo;
    |     ^^^^^^^^
 
-error: aborting due to previous error
+error[E0432]: unresolved import `r#extern`
+  --> $DIR/keyword-extern-as-identifier-use.rs:1:5
+   |
+LL | use extern::foo;
+   |     ^^^^^^ maybe a missing crate `r#extern`?
 
+error: aborting due to 2 previous errors
+
+For more information about this error, try `rustc --explain E0432`.

--- a/src/test/ui/resolve/raw-ident-in-path.rs
+++ b/src/test/ui/resolve/raw-ident-in-path.rs
@@ -1,0 +1,5 @@
+// Regression test for issue #63882.
+
+type A = crate::r#break; //~ ERROR cannot find type `r#break` in module `crate`
+
+fn main() {}

--- a/src/test/ui/resolve/raw-ident-in-path.stderr
+++ b/src/test/ui/resolve/raw-ident-in-path.stderr
@@ -1,0 +1,9 @@
+error[E0412]: cannot find type `r#break` in module `crate`
+  --> $DIR/raw-ident-in-path.rs:3:17
+   |
+LL | type A = crate::r#break;
+   |                 ^^^^^^^ not found in `crate`
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0412`.

